java/gazelle/resolve.go

```diff
@@ -105,48 +103,34 @@ func (jr Resolver) Resolve(c *config.Config, ix *resolve.RuleIndex, rc *repo.Rem
 			continue
 		}
 
-		addDep(dep.String(), imp)
+		deps.Add(dep)
 	}
 
-	if len(deps) > 0 {
-		var labels []label.Label
-		for key := range deps {
-			l, err := label.Parse(key)
-			if err != nil {
-				jr.lang.logger.Fatal().Str("key", key).Err(err).Msg("cannot parse label")
-			}
-
+	if deps.Len() > 0 {
+		var exprs []build.Expr
+		for _, l := range deps.SortedSlice() {
 			if l.Relative && l.Name == from.Name {
 				continue
 			}
-
-			labels = append(labels, l)
-		}
-
-		sort.Slice(labels, func(i, j int) bool {
-			if labels[i].Relative {
-				if labels[j].Relative {
-					return labels[i].String() < labels[j].String()
-				}
-
-				return true
-			}
-
-			if labels[j].Relative {
-				return false
-			}
-
-			return labels[i].String() < labels[j].String()
-		})
-
-		var exprs []build.Expr
-		for _, l := range labels {
 			exprs = append(exprs, &build.StringExpr{Value: l.String()})
 		}
 		r.SetAttr("deps", exprs)
 	}
 }
 
+func labelLess(l, r label.Label) bool {
+	if l.Relative {
+		if r.Relative {
+			return l.String() < r.String()
+		}
+		return true
+	}
+	if r.Relative {
```

Comment:
It's weird, but couldn't you have someone use the fully qualified label to refer to something in the same package? And doing so would fail this test.

Reply:
That's fine here, this is just for sorting, where there isn't really a concept of "the current package" - if people want to canonicalise labels so that in-package ones are relative, they should do so before trying to sort.